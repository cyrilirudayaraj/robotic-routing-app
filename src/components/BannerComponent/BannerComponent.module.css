.banner {
  position: absolute;
  bottom: -7%;
  left: 50%;
  transform: translate(-50%, -50%);
  border-radius: 2px;
  min-width: 20%;
  min-height: 15%;
  box-shadow: 0px 4px 8px -2px rgba(9, 30, 66, 0.25),
    0px 0px 0px 1px rgba(9, 30, 66, 0.08);
}

.header {
  padding: 4%;
  background-color: whitesmoke;
}

.info {
  padding: 4%;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
}

.distance,
.angle {
  display: flex;
  min-width: 100%;
  justify-content: space-between;
}
