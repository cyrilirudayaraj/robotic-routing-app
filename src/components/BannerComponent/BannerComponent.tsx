import React from "react";
import styles from "./BannerComponent.module.css";
import { BannerComponentProps } from "../../intefaces/BannerComponentProps";

export const BannerComponent: React.FC<BannerComponentProps> = (props: any) => {
  return (
    <div className={styles.banner}>
      <div className={styles.header}>
        {`Line ${Number(props.selectedElement?.id) + 1}`}
      </div>
      <div className={styles.info}>
        <div className={styles.distance}>
          <div className={styles.label1}>Length(cms)</div>
          <div role="value1" className={styles.value1}>
            {props.selectedElement?.distance}
          </div>
        </div>
        <div className={styles.angle}>
          <div className={styles.label2}>Angle(degrees)</div>
          <div role="value2" className={styles.value2}>
            {props.selectedElement?.angle}
          </div>
        </div>
      </div>
    </div>
  );
};
