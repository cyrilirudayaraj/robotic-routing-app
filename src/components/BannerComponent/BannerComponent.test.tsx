import { render, screen } from "@testing-library/react";
import { BannerComponent } from "./BannerComponent";

describe("BannerComponent", () => {
  const mockSelectedElement = {
    id: 1,
    distance: 10,
    angle: 45,
  };
  const defaultProps: any = {
    selectedElement: mockSelectedElement,
  };
  it("renders banner component with correct data", () => {
    const { container } = render(<BannerComponent {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
    // Check if the header contains the correct text
    expect(
      screen.getByText(`Line ${mockSelectedElement.id + 1}`).textContent
    ).toEqual(`Line ${mockSelectedElement.id + 1}`);
    expect(Number(screen.getByRole("value1").textContent)).toEqual(
      mockSelectedElement.distance
    );
    expect(Number(screen.getByRole("value2").textContent)).toEqual(
      mockSelectedElement.angle
    );
  });
});
