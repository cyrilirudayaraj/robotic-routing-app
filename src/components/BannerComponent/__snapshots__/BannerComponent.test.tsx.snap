// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerComponent renders banner component with correct data 1`] = `
<div>
  <div>
    Line 2
  </div>
  <div>
    <div>
      <div>
        Length(cms)
      </div>
      <div
        role="value1"
      >
        10
      </div>
    </div>
    <div>
      <div>
        Angle(degrees)
      </div>
      <div
        role="value2"
      >
        45
      </div>
    </div>
  </div>
</div>
`;
